### PR TITLE
Fixes: Debounce and throttling added

### PR DESCRIPTION
**Pull request by Dev-AkshatGupta**
Search functionality with Debounce added 
Please review this pr thanks.

### CODE REVIEW

File: src/Pages/BookMarkPage/bookMarkPage.css [hunk 1]
```diff
@@ -27,6 +27,7 @@ max-width:573px ;}
   align-items: center;
   justify-content: flex-start;
   top: 0;
+  /* left:0; */
```
Comment (b0llu) [inline src/Pages/BookMarkPage/bookMarkPage.css]: remove commented code

**Review by b0llu — APPROVED** [review]
Clean code. You can merge this PR

File: src/Pages/BookMarkPage/bookMarkPage.css (hunk 1 above)
Comment (Dev-AkshatGupta) [reply src/Pages/BookMarkPage/bookMarkPage.css]: Thanks for reviewing sure removing it

File: src/Utilities/debounce.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  let timeoutId;
+  return function (...args) {
+    const context = this;
+    clearTimeout(timeoutId);
+    timeoutId = setTimeout(() => callbackFn.apply(context, args), delay);
+  };
+};
+export const throttling = (callbackFn, delay) => {
+  let flag = true;
+  return function (...args) {
```
Comment (Developers-codz) [inline src/Utilities/debounce.js:11]: appreciation - for using both debounce and throttling

**Review by Developers-codz — APPROVED** [review]
Code looks clean, nothing to suggest you can merge

**Pull request merged**